### ISSUE TITLE
Update OS-detection checks to support analyzers

### ISSUE THREAD

**Issue by rainersigwald** [dotnet/msbuild#6376]
Modern .NET knows about platform-specific APIs and can issue nice compile-time warnings for them. We should update our own usages to either directly use the new nice APIs or at least be compatible with them.

https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1416

**Comment by BenVillalobos** [comment]
Yep, this should be fixed by https://github.com/dotnet/msbuild/pull/6148

**Comment by rainersigwald** [comment]
There are a few `#pragma warning disable CA1416` in the codebase that we should remove as part of this.
